src/snowflake/snowpark/types.py

```diff
@@ -40,12 +41,17 @@ def __repr__(self):
 
     def is_primitive(self):
         return True
-
+    
+    def fill_ast(self, ast: proto.SpDataType) -> None:
+        raise NotImplementedError(f"{self.__class__.__name__} does not have an equivalent SpDataType yet")
```

Comment:
this is a base method I assume? Maybe let's not use NotImplementedError, but instead ValueError and what to implement.

Reply:
you can also use https://docs.python.org/3/library/abc.html

Reply:
There are some intermediary types like `_AtomicType(DataType)` that should not be required to implement `fill_ast` I think so I will raise a `ValueError` instead.